R/aedseo.R

```diff
@@ -79,6 +79,16 @@ aedseo <- function(
       # TODO: #10 Include negative.binomial regressions. @telkamp7
     ),
     na_fraction_allowed = 0.4) {
+  # Check input arguments
+  coll <- checkmate::makeAssertCollection()
+  checkmate::assert_data_frame(tsd)
+  checkmate::assert_class(tsd, "aedseo_tsd")
+  checkmate::assert_names(names(tsd), identical.to = c("time", "observed"))
+  checkmate::assert_numeric(c(level, na_fraction_allowed), len = 2,
+                            lower = 0, upper = 1, add = coll)
+  checkmate::assert_integerish(c(k, disease_threshold), len = 2)
```

Comment:
It is quire clever to condense these to two lines instead of four, but I think it is more clear to the user if they are just repeated lines with different variables. (in my opinion).